### DISCUSSION TITLE
How about adding `rayon` and multithreading where are these needs?

### DISCUSSION THREAD

**Discussion by Jerrody** [kimlimjustin/xplorer#234] [Ideas]
For example:
```rust
pub async fn get_files_in_directory(dir: &Path) -> Result<Vec<String>, String>
```
It iterates over the path and it's not so much damage to the performance, but if we have 1000 files and so on? And in this case would be pretty good if we had `rayon`.

**Comment by kimlimjustin** [comment]
I'm not really familiar with `rayon` but any PR will be appreciated
